oauth.rst

```diff
@@ -1,131 +1,129 @@
-Partner Integrations: Public Documentation
-=========================================
+DOCK OAuth Integration Guide
+============================
 
-The OAuth 2.0 spec defines four roles:
+Initial Setup
+-------------
 
-- Resource owner
-- Resource server
-- Authorization server
-- Client
+Any new Partner application needs to be first set up with DOCK manually before any authentication or data exchange services can be used. Following are the one time set up steps for partners:
 
-The following document will try to explain how Dock approaches its OAuth2 implementation.
+- Provide basic info like your application's name, logo and description.
+- Provide these URLs: homepage, privacy policy and at least one callback URI. 
+The above information, except for the callback URIs, will be shown to the User at the Authorization Page. 
 
-The User
---------
-The OAuth 2.0 spec refers to the user as the "resource owner". The resource owner is the person who is giving access to some portion of their account. In our case this is a user in the Dock App. The resources in our case is data: personal information, contacts, work experience, etc. Any system that wants to act on behalf of the user must first get permission from them.
+- As a result of registering, you will be given a ``client_id`` and a ``client_secret`` that you will need to use to authenticate your requests to our API.  Please be aware that we do not store your ``client_secret``. You need to save it somewhere safe as soon as you get it since we cannot retrieve it back for you if you lose it. In that event a new set of client id/secret will have to be created.
 
-The Resource Server
--------------------
-What you would typically think of as the main API is called the "resource server". In our case this is the ``dockapi``: the server that contains the user's information that is being accessed by the third-party application.
 
-The Authorization Server
-------------------------
-The authorization server is what the user interacts with when an application is requesting access to their account. In our case this is the ``dock-auth`` service.
+Basic Client Integration Flow
+=============================
 
-The Client
-----------
-The Client is the 3rd party app that is attempting to act on the user's behalf, or access the user's resources. Before the Client can access the user's account, it needs to obtain permission from the user.
+Quick details
+-------------
+Following is the simplified version of the flow before diving into details: 
 
-Client Registration
--------------------
-When developers register their applications they need to provide some basic info:
+- Clients/Partner request an ``authorization_code`` grant type by redirecting the user to the Dock Authorization Page. 
```

Comment:
Do we need to have "Clients/Partner" here? In that case I'd use plural for partner. But I'd suggest we keep just Client since that's the wording that the document uses in general.